frontend/src/app/components/ProfileStatistics/index.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+
+import { TextSpan, StatisticsTable } from '../../../styles/blocks'
+import { StatsRow } from './StatsRow'
+
+export const ProfileStatistics = props => {
+  const {
+    rankings: { ranking, toNextRank, toPrevRank },
+    statistics: {
+      bestDay, worstDay,
+      mostFrequentTeammate, leastFrequentTeammate, mostFrequentOpponent, leastFrequentOpponent,
+      mostSuccessTeammate, leastSuccessTeammate, mostSuccessOpponent, leastSuccessOpponent,
+    },
+  } = props
+
+  const rows = [
+    [
+      createRankRowObject('To Next Rank', toNextRank, '+', toNextRank <= 20),
+      createRankRowObject('To Prev Rank', toPrevRank, '-', toPrevRank > 20),
+    ], [
+      createDateRowObject('Best Day', bestDay || {}),
+      createDateRowObject('Worst Day', worstDay || {}),
+    ], [
+      createRowObject('Most Played With', mostFrequentTeammate, 'x', true),
+      createRowObject('Least Played With', leastFrequentTeammate, 'x', false),
+    ], [
+      createRowObject('Most Played Against', mostFrequentOpponent, 'x', true),
+      createRowObject('Least Played Against', leastFrequentOpponent, 'x', false),
+    ], [
+      createRowObject('Most Won With', mostSuccessTeammate, 'x', true),
+      createRowObject('Most Lost With', leastSuccessTeammate, 'x', false),
+    ], [
+      createRowObject('Most Won Over', mostSuccessOpponent, 'x', true),
+      createRowObject('Most Lost To', leastSuccessOpponent, 'x', false),
+    ],
+  ]
+
+  return (
+    <>
+      <TextSpan>Ranking: {ranking}</TextSpan>
+      <StatisticsTable>
+        <tbody>
+          {rows.map((row, index) => (
+            <tr key={index}>
+              {row.map((node, key) => (
+                <StatsRow key={[index, key].join('_')}
+                  text={node.text}
+                  player={node.player}
+                  score={node.score}
+                  positive={node.positive}
+                  altText={node.altText} />
+              ))}
+            </tr>
+          ))}
+        </tbody>
+      </StatisticsTable>
+    </>
+  )
+}
+
+const createRowObject = (text, obj, sign, positive) => ({
```

Comment:
My preference would be several separate simple components instead of one generic StatsRow with those three props creators... let's ask someone else as well. @novellizator @petrk-salsita what do you think?

Reply:
I agree that decomposing into reusable components would probably be cleaner 👍

Reply:
Good idea, done [here](https://github.com/salsita/foosball-rating/pull/114/commits/48d97a59bd3178627233fc0798f8d61a47a09c3b#diff-744525510377a0aca19d17cf4dc35116), [here](https://github.com/salsita/foosball-rating/pull/114/commits/48d97a59bd3178627233fc0798f8d61a47a09c3b#diff-e1417bef467b606501c1db87ed6132c1) and below